docs/DeveloperGuide.md

```diff
@@ -1882,15 +2069,98 @@ testers are expected to do more *exploratory* testing.
+      1. F.A.K.E.J.A.R.V.I.S. has been initialized.
+   2. Test case: Delete a part of any `.json` file in `data` folder, or add some random symbols (e.g `@#$%^&*()`. Then relaunch F.A.K.E.J.A.R.V.I.S.<br>
+      Expected: F.A.K.E.J.A.R.V.I.S. recognizes the corrupted data file. The corresponding tabs will be empty (i.e if `tasklist.json` is corrupted, `Task List` will be empty).
+
+2. Dealing with missing data files
+   1. Prerequisite:
+      1. F.A.K.E.J.A.R.V.I.S. has been initialized.
+   2. Test case: Delete the whole `.json` file in `data` folder. Then relaunch F.A.K.E.J.A.R.V.I.S.<br>
+      Expected: F.A.K.E.J.A.R.V.I.S. recognizes the missing data file and add a sample data file inplace of the missing file.
+
+
+## **Appendix: Planned Enhancements**
+
+1. **Text Overflow Handling**
+   * Current Implementation: Currently, long emails, long task descriptions or names may extend beyond the allocated space, making it challenging for users to view the complete information. The enhancement will truncate or wrap the text appropriately, maintaining a clean and organized display.
+   * Enhancement: Implement a mechanism to handle text overflow, to ensure that lengthy text entries do not disrupt the user interface.
+   * Reason: To improve the readability and visual clarity of GUI.
+   * Suggested Fixes:
+     * Have character limits on certain fields (e.g `Email`, `Name`, `TaskName`).
+     * Have a character length checker that auto wraps the text when it "overflowed".
+   
+<br>
+   
+2. **Phone Number Validation**
+   * Current Implementation: Currently, any number that is longer than 3 digits are considered valid. It doesn't allow for special characters too.
+   * Enhancement: Introduce a validation mechanism for phone numbers to ensure that entered phone numbers adhere to a specified format. (e.g SG phone number starts with 8 or 9 and have)
+   * Reason: To enhance the accuracy of the phone numbers by validating them according to a standard format.
+   * Suggested Fixes:
+     * Have a validation regex to check for starting number = 8 or 9 (For SG phone numbers).
+     * Allow some special characters such as `+` to enable the adding of country codes in the phone number field.
 
-  1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+<br>
 
-1. _{ more test cases …​ }_
+3. **CSV File to Json Conversion and Vice Versa**
+   * Current Implementation: Currently, data can only be added/edited via the F.A.K.E.J.A.R.V.I.S. app, or directly editing the `.json` file.
+   * Enhancement: Implement the functionality to convert data to and from CSV files, enabling users to conveniently import/export data without manual editing.
+   * Reason: To enhance user convenience and data interchangeability, allowing for seamless integration with external applications that support CSV formats.
+   * Suggested Fixes:
+     * Have a `convertCsvToJson` and `convertJsonToCsv` method to convert the files accordingly.
```

Comment:
I think instead of "convert", possibly just abstract out the conversion and just allow users to download either json or csv files?